sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp

```diff
@@ -0,0 +1,772 @@
+
+    /**
+     * @brief Enables using the equal operator for key curve.
+     *
+     * @param other A key curve to be compared.
+     */
+    bool operator==(const CertificateContentType& other) const noexcept
+    {
+      return m_value == other.m_value;
+    }
+
+    /**
+     * @brief Get the string value of the key curve.
+     *
+     */
+    std::string const& ToString() const { return m_value; }
+
+    /**
+     * @brief Get the NIST P-256 elliptic curve, AKA SECG curve SECP256R1.
+     *
+     * @remark For more information, see
+     *  <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateContentType Pkcs12;
+
+    /**
+     * @brief Get the SECG SECP256K1 elliptic curve.
+     *
+     * @remark For more information, see
+     * <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateContentType Pem;
+  };
+
+  /**
+   * @brief Supported usages of a certificate key.
+   *
+   */
+  class CertificateKeyUsage final {
+  private:
+    std::string m_value;
+
+  public:
+    /**
+     * @brief Construct a new Key Curve Name object.
+     *
+     * @param value The string value of the instance.
+     */
+    explicit CertificateKeyUsage(std::string value)
+    {
+      if (value.empty())
+      {
+        throw std::invalid_argument("The value for the curve name can not be empty");
+      }
+      m_value = std::move(value);
+    }
+
+    /**
+     * @brief Construct a default key curve.
+     *
+     */
+    CertificateKeyUsage() = default;
+
+    /**
+     * @brief Enables using the equal operator for key curve.
+     *
+     * @param other A key curve to be compared.
+     */
+    bool operator==(const CertificateKeyUsage& other) const noexcept
+    {
+      return m_value == other.m_value;
+    }
+
+    /**
+     * @brief Get the string value of the key curve.
+     *
+     */
+    std::string const& ToString() const { return m_value; }
+
+    /**
+     * @brief Get the NIST P-256 elliptic curve, AKA SECG curve SECP256R1.
+     *
+     * @remark For more information, see
+     *  <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyUsage DigitalSignature;
+
+    /**
+     * @brief Get the SECG SECP256K1 elliptic curve.
+     *
+     * @remark For more information, see
+     * <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyUsage AutoRenew;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyUsage EmailContacts;
```

Comment:
Missing comment, as well as a large number of values here. See .NET's for examples.